From 6d54ca971d490eb5a93224ad45a59c11004f93d6 Mon Sep 17 00:00:00 2001
From: JosephMcc <mccullarjoseph@gmail.com>
Date: Thu, 6 Nov 2014 03:24:00 -0800
Subject: [PATCH] re-enable the right click context menu on nemo toolbar
 navigation buttons and remove unused nemo-toolbar-ui.xml file

---
 src/nemo-navigation-action.c | 50 +++++++++++++-------------------------------
 src/nemo-toolbar-ui.xml      | 22 -------------------
 2 files changed, 15 insertions(+), 57 deletions(-)
 delete mode 100644 src/nemo-toolbar-ui.xml

diff --git a/src/nemo-navigation-action.c b/src/nemo-navigation-action.c
index 8ef85d8..0fe152b 100644
--- a/src/nemo-navigation-action.c
+++ b/src/nemo-navigation-action.c
@@ -233,38 +233,20 @@ tool_button_release_cb (GtkButton *button,
         return FALSE;
 }
 
-static GtkWidget *
-get_actual_button (GtkToolButton *tool_button)
-{
-	GList *children;
-	GtkWidget *button;
-
-	g_return_val_if_fail (GTK_IS_TOOL_BUTTON (tool_button), NULL);
-
-	children = gtk_container_get_children (GTK_CONTAINER (tool_button));
-	button = GTK_WIDGET (children->data);
-
-	g_list_free (children);
-
-	return button;
-}
-
 static void
 connect_proxy (GtkAction *action,
                GtkWidget *proxy)
 {
-        GtkToolButton *tool;
-        GtkWidget *button;
+    GtkWidget *button;
 
-	if (GTK_IS_TOOL_BUTTON (proxy)) {
-                tool = GTK_TOOL_BUTTON (proxy);
-                button = get_actual_button (tool);
+	if (GTK_IS_BUTTON (proxy)) {
+        button = GTK_BUTTON (proxy);
 
-                g_signal_connect (button, "button-press-event",
-                                  G_CALLBACK (tool_button_press_cb), action);
-                g_signal_connect (button, "button-release-event",
-                                  G_CALLBACK (tool_button_release_cb), action);
-        }
+        g_signal_connect (button, "button-press-event",
+                          G_CALLBACK (tool_button_press_cb), action);
+        g_signal_connect (button, "button-release-event",
+                          G_CALLBACK (tool_button_release_cb), action);
+    }
 
 	(* GTK_ACTION_CLASS (nemo_navigation_action_parent_class)->connect_proxy) (action, proxy);
 }
@@ -273,20 +255,18 @@ static void
 disconnect_proxy (GtkAction *action,
                   GtkWidget *proxy)
 {
-        GtkToolButton *tool;
-        GtkWidget *button;
+    GtkWidget *button;
 
-	if (GTK_IS_TOOL_BUTTON (proxy)) {
-                tool = GTK_TOOL_BUTTON (proxy);
-                button = get_actual_button (tool);
+	if (GTK_IS_BUTTON (proxy)) {
+        button = GTK_BUTTON (proxy);
 
-                /* remove any possible timeout going on */
-                unschedule_menu_popup_timeout (NEMO_NAVIGATION_ACTION (action));
+        /* remove any possible timeout going on */
+        unschedule_menu_popup_timeout (NEMO_NAVIGATION_ACTION (action));
 
 		g_signal_handlers_disconnect_by_func (button,
-                                                      G_CALLBACK (tool_button_press_cb), action);
+                                              G_CALLBACK (tool_button_press_cb), action);
 		g_signal_handlers_disconnect_by_func (button,
-                                                      G_CALLBACK (tool_button_release_cb), action);
+                                              G_CALLBACK (tool_button_release_cb), action);
 	}
 
 	(* GTK_ACTION_CLASS (nemo_navigation_action_parent_class)->disconnect_proxy) (action, proxy);
diff --git a/src/nemo-toolbar-ui.xml b/src/nemo-toolbar-ui.xml
deleted file mode 100644
index ab79c74..0000000
--- a/src/nemo-toolbar-ui.xml
+++ /dev/null
@@ -1,22 +0,0 @@
-<ui>
-<toolbar name="Toolbar">
-	<toolitem name="Back" action="Back"/>
-	<toolitem name="Forward" action="Forward"/>
-	<toolitem name="Up" action="Up"/>
- 	<toolitem name="Reload" action="Reload"/>
-	<separator name="AfterNavigation" />        
-        <toolitem name="Home" action="Home"/>
-        <toolitem name="Computer" action="Computer"/>
-	<separator name="BeforeLocation" />	
-</toolbar>
-<toolbar name="SecondaryToolbar">		
-	<toolitem name="Edit Location" action="Toggle Location Button"/>
-        <toolitem name="Open in Terminal" action="OpenInTerminal"/>
-        <toolitem name="New Folder" action="New Folder"/>
-        <toolitem name="Search" action="Search"/>
-        <separator name="ViewTypeSeparator" />
-        <toolitem name="Icon View" action="IconView"/>
-        <toolitem name="List View" action="ListView"/>
-        <toolitem name="Compact View" action="CompactView"/>    
-</toolbar>
-</ui>
